tests/unit/test_client.py

```
@@ -0,0 +1,357 @@
+class TestNetworkClient:
+    """Test suite for NetworkClient functionality."""
+    
+    def test_network_client_initialization(self):
+        """Test NetworkClient initializes correctly."""
+        client = NetworkClient()
+        assert client is not None
+        
+    def test_network_client_with_timeout(self):
+        """Test NetworkClient initializes with custom timeout."""
+        client = NetworkClient(timeout=60)
+        assert client.timeout == 60
```

Ruff (0.11.9)

Magic value used in comparison, consider replacing `60` with a constant variable

(PLR2004)